src/Items/ItemLilypad.h

```diff
@@ -57,51 +51,146 @@ class cItemLilypadHandler :
 			}
 
 			virtual bool OnNextBlock(int a_CBBlockX, int a_CBBlockY, int a_CBBlockZ, BLOCKTYPE a_CBBlockType, NIBBLETYPE a_CBBlockMeta, eBlockFace a_CBEntryFace) override
-			{
-				if (IsBlockWater(a_CBBlockType))
-				{
-					if ((a_CBBlockMeta != 0) || (a_CBEntryFace == BLOCK_FACE_NONE))  // The hit block should be a source. The FACE_NONE check is clicking whilst submerged
-					{
-						return false;
-					}
-					AddFaceDirection(a_CBBlockX, a_CBBlockY, a_CBBlockZ, BLOCK_FACE_YP);  // Always place pad at top of water block
-					if (
-						!IsBlockWater(a_CBBlockType) &&
-						cBlockInfo::FullyOccupiesVoxel(a_CBBlockType)
-						)
-					{
-						// Can't place lilypad on air / in another block!
-						return true;
-					}
+			{	
+				if((((a_CBBlockType == E_BLOCK_STATIONARY_WATER) || (a_CBBlockType == E_BLOCK_WATER)) && (a_CBBlockMeta == 0))
+				|| (a_CBBlockType == E_BLOCK_ICE) || (a_CBBlockType == E_BLOCK_FROSTED_ICE))
+				{	
```

Comment:
This is now an unreadable mess. With a condition this long, split it onto multiple lines with *each* of the same-level conditions on a separate lines.

What's wrong with `IsBlockWater`?

Reply:
I can try IsBlockWater, but the old lily pad code wasn't using it, and I assumed it was for a good reason.

Reply:
B... b... but it's right there, line 61 of the original code.